core/src/main/java/org/axonframework/commandhandling/model/inspection/AnnotatedChildEntity.java

```diff
@@ -71,7 +73,9 @@ public void publish(EventMessage<?> msg, P declaringInstance) {
         if (forwardEvents) {
             Iterable<C> targets = eventTargetResolver.apply(msg, declaringInstance);
             if (targets != null) {
-                targets.forEach(target -> this.entityModel.publish(msg, target));
+                Collection<C> copy = new ArrayList<>();
```

Comment:
I like the framework being lenient about this. Perhaps we could even detect these implementations and then not make a copy. It's viable for an event to change the contents of a collection.
So - if Concurrent Impl -> iterate. If List, copy. Else, iterate and add to new list.